### ISSUE TITLE
Provide content types

### ISSUE THREAD

**Issue by Guite** [zikula-modules/MediaModule#2]
It would be nice to have some content types for integration.
1. A single medium
   I've seen there is a Scribite integration already. A content type offers more sophisticated possibilities regarding positioning and styling though (there are use cases where a dedicated content element is more useful).
2. A collection
   In SimpleMedia there is a content type for including a list of media items filtered by a category. I do not know yet if this module supports categories, too, but there are collections.
   It should be possible to add a list of media items based on a selected template. This allows for including galleries, sliders and other things into Content pages.

**Comment by cmfcmf** [comment]
I played around with content types for a bit but can't currently make it work for two reasons:
1. Content expects content type class names to be "Module_ContentType_Name" and doesn't support namespaced content types.
2. Content forces content types to use Smarty + Zikula Forms + static methods. 

I'll postpone this until Content has been updated to support service-based content types and is decoupled from Zikula forms.

**Comment by Guite** [comment]
Fine for me. Just wanted to ensure that this feature will not be forgotten. We should aim for a Content rewrite in 2016 (being aware that this is currently blocked by the generator being updated to Twig and Symfony Forms ;-)).

**Comment by rallek** [comment]
Content plugin is very important for me. I am using it intensively on my pggo site. Here it would be needed similar to media share

**Comment by MrMontesa** [comment]
Just stubled upon this issue and would like to vote up +1.
MediaModule is the defacto standard for image handling for 1.4.x going forward. And Content is the main texting tool. I would love to include
a) single image into my Content pages
b) have 4-5 pics teaser thumbnails linking to a collection from within Content

Previously I used [MediaShare](https://github.com/zikula-modules/Mediashare) which provides these 2 features. link to [content_types_api](https://github.com/zikula-modules/Mediashare/tree/master/pncontenttypesapi)

**Comment by craigh** [comment]
the problem is that Content will not run on Core-2.0

**Comment by rallek** [comment]
@cmfcmf  told during the Camp Zikula 2016 Media Module will only have code for 1.4/2.0 core to be a valid example for other module developers. The content code would be possible but legacy.

But there could be a solution until Content is transfered into the new century: We should ask Christian to provide blocks. So we would be able to use the specified blocks within Content. That is not comfortable but a work around.

@MrMontesa: please create tickets with the block description you might need. I am shure Christian will try to implement these.

**Comment by Guite** [comment]
Currently working on Content 5.0 for Zikula 2. Going to reference docs and examples for new content types implementation as soon as the interface is fixed.

**Comment by Guite** [comment]
Not entirely finished yet, but here is some information:

https://github.com/zikula-modules/Content/tree/5.0.0#implementing-custom-content-types
https://github.com/zikula-modules/Content/blob/5.0.0/src/modules/Zikula/ContentModule/ContentTypeInterface.php
https://github.com/zikula-modules/Content/blob/5.0.0/src/modules/Zikula/ContentModule/AbstractContentType.php
https://github.com/zikula-modules/Content/tree/5.0.0/src/modules/Zikula/ContentModule/ContentType
https://github.com/zikula-modules/Content/tree/5.0.0/src/modules/Zikula/ContentModule/ContentType/Form/Type
https://github.com/zikula-modules/Content/tree/5.0.0/src/app/Resources/ZikulaContentModule/views/ContentType

**Comment by Guite** [comment]
Content 5 beta has been released. The new ContentTypeInterface can be considered as stable now.
